File: API/Entities/Rating.cs
using System;

namespace API.Entities;

public class Rating
{
    public int Id { get; set; }
    public int LenderId { get; set; }
    public User Lender { get; set;}
    public int BorrowerId { get; set; }
    public User Borrower { get; set;}

    public int LoanApplicationId { get; set;}
    public LoanApplication LoanApplication { get; set;}
    public int Score { get; set; } // Score from 1 to 5
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: client/src/app/_models/rating.ts
import { LoanApplication } from "./loanApplication";
import { User } from "./user";

export interface Rating {
    id: number;
    lenderId: number;
    lender: User;
    borrowerId: number;
    borrower: User;
    loanApplicationId: number;
    loanApplication: LoanApplication;
    score: number;
    createdAt: string;
    updatedAt: string;
}
